compiler/test/dotty/tools/dotc/reporting/TestReporter.scala

```diff
@@ -144,4 +152,16 @@ object TestReporter {
     }
     rep
   }
+
+  def lastRunFailedTests: Option[List[String]] =
+    Option.when(Properties.rerunFailed)(
+      if (failedTestsFile.exists() && failedTestsFile.isFile)
+      then java.nio.file.Files.readAllLines(failedTestsFile.toPath).asScala.toList
+      else List.empty
+    )
```

Comment:
```suggestion
    Option.when(
      Properties.rerunFailed &&
      failedTestsFile.exists() &&
      failedTestsFile.isFile
    )(
java.nio.file.Files.readAllLines(failedTestsFile.toPath).asScala.toList
    )
```

Reply:
The absence of `failedTestsFile` should also mean that we need to run all the tests

Reply:
done